### PR TITLE
[#3] Create base application layout

### LINKED ISSUE #3: Create base application layout
## Why

Describe the big picture of the feature and why it is needed.

## Acceptance Criteria

- User should see the blank page with the background when accessing the site

## Resources

- https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App

### PR DESCRIPTION
**Pull request by rosle**
https://github.com/rosle/react-nimble-survey/issues/3

## What happened 👀

- Add new route - `/sign_in`
- Add base layout styles

  - **Default Layout**: For the main application pages
  - **Auth Layout**: For the authentication pages
  
- Set up localization namespaces - `shared` and `auth`

### @import

For Styling, I decided to revert back to `@import` instead of `@use` and `forward`. Because when we use `@use`/`@forward`, it will apply on that specific file only. So imagine if we want to use variables on file A and file B, we need to `@use`/`@forward` on both files.

### React Helmet Async

I use [React Helmet](https://github.com/nfl/react-helmet) to modify the layout className. But there are errors with the new version of React. See issue **548** on the Repo

Screen Shot 2565-08-10 at 10 58 45

https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

So now, need to move to use [React Helmet Async](https://github.com/staylor/react-helmet-async) instead

## Proof Of Work 📹

- Default Layout: https://62f0ef1bf3c82a5549ad4033--react-nimble-survey-staging.netlify.app

Screen Shot 2565-08-08 at 18 16 24

- Auth Layout: https://62f0ef1bf3c82a5549ad4033--react-nimble-survey-staging.netlify.app/sign_in

Screen Shot 2565-08-08 at 18 16 11

### CODE REVIEW

File: src/assets/stylesheets/functions/_sizing.scss [hunk 1]
```diff
@@ -20,7 +20,7 @@
     @if ($value == 0 or $value == auto) {
       $list: append($list, $value);
     } @else {
-      $rem-value: ($value / $base-font-size) + rem;
+      $rem-value: calc($value / $base-font-size) + rem;
```
Comment (rosle) [inline src/assets/stylesheets/functions/_sizing.scss:23]: To fix the warning ⚒️ 

```
Deprecation Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($value, $base-font-size) or calc($value / $base-font-size)

More info and automated migrator: https://sass-lang.com/d/slash-div
```

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import logo from 'assets/images/logo.svg';
+import DefaultLayout from 'components/Layout/Default';
+
+const HomeScreen = (): JSX.Element => {
```
Comment (hanam1ni) [inline src/screens/Home/index.tsx]: We can omit the return type here.
```suggestion
const HomeScreen = () => {
```

**Review by hanam1ni — APPROVED** [review]
One minor comment

File: src/screens/Home/index.tsx (hunk 2 above)
Comment (rosle) [reply src/screens/Home/index.tsx]: Thank you! Updated at [38f1b30](https://github.com/rosle/react-nimble-survey/pull/11/commits/38f1b30bcab5bed5bb82f7b933a22170e33f3277) 🚀

**Comment by Lahphim** [comment]
> 🚀 Deployed on https://62fb000b2013e72cc17b5eed--react-nimble-survey-staging.netlify.app

I cannot access to this preview 🫣

**Review by Lahphim — APPROVED** [review]

**Comment by rosle** [comment]
> https://62fb000b2013e72cc17b5eed--react-nimble-survey-staging.netlify.app

How does it display on your end, @Lahphim? Could you provide me the screenshot? 😆🙏

**Pull request merged**